fuel-tx/src/transaction/types/input/coin.rs

```diff
@@ -135,216 +130,64 @@ where
-        } else {
-            buf
-        };
-
-        if let Some(predicate_data) = predicate_data.as_field() {
-            bytes::store_raw_bytes(buf, predicate_data.as_slice())?;
-        };
-
-        Ok(serialized_size)
-    }
-}
-
-#[cfg(feature = "std")]
-impl<Specification> std::io::Write for Coin<Specification>
-where
-    Specification: CoinSpecification,
-{
-    fn write(&mut self, full_buf: &[u8]) -> std::io::Result<usize> {
-        use fuel_types::bytes::Deserializable;
-        type S = CoinSizes;
-        const LEN: usize = CoinSizes::LEN;
-        let buf: &[_; LEN] = full_buf
-            .get(..LEN)
-            .and_then(|slice| slice.try_into().ok())
-            .ok_or(bytes::eof())?;
-
-        let mut n = LEN;
-
-        let utxo_id = UtxoId::from_bytes(&buf[S::LAYOUT.utxo_id.range()])?;
-        self.utxo_id = utxo_id;
-
-        let owner = bytes::restore_at(buf, S::layout(S::LAYOUT.owner));
-        let owner = owner.into();
-        self.owner = owner;
-
-        let amount = bytes::restore_number_at(buf, S::layout(S::LAYOUT.amount));
-        self.amount = amount;
-
-        let asset_id = bytes::restore_at(buf, S::layout(S::LAYOUT.asset_id));
-        let asset_id = asset_id.into();
-        self.asset_id = asset_id;
-
-        let tx_pointer = TxPointer::from_bytes(&buf[S::LAYOUT.tx_pointer.range()])?;
-        self.tx_pointer = tx_pointer;
-
-        let witness_index = bytes::restore_u8_at(buf, S::layout(S::LAYOUT.witness_index));
-        if let Some(witness_index_field) = self.witness_index.as_mut_field() {
-            *witness_index_field = witness_index;
-        }
-        let maturity = bytes::restore_u32_at(buf, S::layout(S::LAYOUT.maturity)).into();
-        self.maturity = maturity;
-
-        let predicate_gas_used =
-            bytes::restore_number_at(buf, S::layout(S::LAYOUT.predicate_gas_used));
-        if let Some(predicate_gas_used_field) = self.predicate_gas_used.as_mut_field() {
-            *predicate_gas_used_field = predicate_gas_used;
-        }
-
-        let predicate_len =
-            bytes::restore_usize_at(buf, S::layout(S::LAYOUT.predicate_len));
-        let predicate_data_len =
-            bytes::restore_usize_at(buf, S::layout(S::LAYOUT.predicate_data_len));
-
-        let (size, predicate, buf) = bytes::restore_raw_bytes(
-            full_buf.get(LEN..).ok_or(bytes::eof())?,
-            predicate_len,
-        )?;
-        n += size;
-        if let Some(predicate_field) = self.predicate.as_mut_field() {
-            *predicate_field = predicate;
-        }
-
-        let (size, predicate_data, _) =
-            bytes::restore_raw_bytes(buf, predicate_data_len)?;
-        n += size;
-        if let Some(predicate_data_field) = self.predicate_data.as_mut_field() {
-            *predicate_data_field = predicate_data;
+        Coin {
+            utxo_id,
+            owner,
+            amount,
+            asset_id,
+            tx_pointer,
+            witness_index: (),
```

Comment:
Can this line be replaced with `..Default::default()` for consistency with your changes below?

Reply:
79d57179